python/cudf_polars/cudf_polars/experimental/io.py

```diff
@@ -127,9 +127,21 @@ def from_scan(
             column_stats = stats.column_stats.get(ir, {})
             column_sizes: list[int] = []
             for cs in column_stats.values():
-                storage_size = cs.source_info.storage_size
-                if storage_size.value is not None:
-                    column_sizes.append(storage_size.value)
+                element_size = 1
+                if (dtype := ir.schema.get(cs.name)) is not None:
+                    if dtype.id() in (
+                        plc.TypeId.STRING,
+                        plc.TypeId.FLOAT64,
+                        plc.TypeId.INT64,
+                    ):
+                        element_size = 8
```

Comment:
FWIW, polars implements the [small string optimization](https://pola.rs/posts/polars-string-type/) optimization, so this minimum size might not be small enough for small strings :/

Reply:
Thanks for the link.

I don't think we care how Polars stores a string column unless libcudf does the same thing?

That said, the approach used in this PR is meant to be a very simple sanity check, and I'm **very** open to changes. Ideally, we can use a **real** row-group sample to adjust the partitioning scheme. However, I haven't worked out a clean way to do that yet.